### ISSUE TITLE
[v2] Custom certificates not getting stored

### ISSUE THREAD

**Issue by NochEinKamel** [jc21/nginx-proxy-manager#12]
Hi,
thanks for the quick reaction on my previous isse.

After beeing able to upload the certificate without errors the proxy won't come up if I set it up to use my custom ssl certificate.

I manager to steal some time off work (;-)) and debugged the issue so far:

Creating a new certificate calls the create function and in the create function the writeCustomCert function is called. However the certificates are not uploaded yet to the express backend but are transferred in a seperate, later upload request.
The upload request triggers an update (instead of create) but nowhere in the upload/update functions the files are written to the disk.
I've added the following code after the patchAndFetchById call in the update method. After Line 230 here: https://github.com/jc21/nginx-proxy-manager/blob/5ac0e3dc95960301ce331e54b68711c484dbad57/src/backend/internal/certificate.js

```
...
.patchAndFetchById(row.id, data)
.then(certificate => {
  return internalCertificate.writeCustomCert(certificate)
    .then(() => {
      return certificate;
    });
})
.then(saved_row => {
...
```

It works in regards of the privkey and fullchain files in the data directory but I'm not sure if it's the correct outcome on the database side as the file contents are kept and written in the database.
Perhaps a better place would be the cleanMeta function? Or the upload function itself.

p.s. also the writeCustomCert part in the create function is probably obsolete

**Comment by jc21** [comment]
Yep you're absolutely right. The writing of custom certs was in the wrong place. I must have written that part in a hurry.

Version 2.0.2 has the fixes now, I've tested this time :)

The reason for the file content to be saved in the database is that it's the source of truth. The contents of the `/data` folder could be wiped, or not mounted in the docker container and despite being a docker Volume as a persistent fallback, it could still be wiped by accident. Although I haven't included the functionality yet, the idea is that the Nginx config and certificates could be regenerated at any time from the database.

I noticed that the frontend was exposed to the custom certificates file content by accident. I've also fixed that in the 2.0.2 release.

Let me know how you go :)
